src/plugins/part.c

```diff
@@ -779,46 +870,28 @@ BDPartSpec* bd_part_get_part_spec (const gchar *disk, const gchar *part, GError
  * Tech category: %BD_PART_TECH_MODE_QUERY_PART + the tech according to the partition table type
  */
 BDPartSpec* bd_part_get_part_by_pos (const gchar *disk, guint64 position, GError **error) {
-    PedDevice *dev = NULL;
-    PedDisk *ped_disk = NULL;
-    PedPartition *ped_part = NULL;
+    BDPartSpec **parts = NULL;
     BDPartSpec *ret = NULL;
-    PedSector sector = 0;
 
-    dev = ped_device_get (disk);
-    if (!dev) {
-        set_parted_error (error, BD_PART_ERROR_INVAL);
-        g_prefix_error (error, "Device '%s' invalid or not existing", disk);
+    parts = get_disk_parts (disk, TRUE, TRUE, error);
+    if (!parts)
         return NULL;
-    }
 
-    ped_disk = ped_disk_new (dev);
-    if (!ped_disk) {
-        set_parted_error (error, BD_PART_ERROR_FAIL);
-        g_prefix_error (error, "Failed to read partition table on device '%s'", disk);
-        ped_device_destroy (dev);
-        return NULL;
-    }
+    for (BDPartSpec **parts_p = parts; *parts_p; parts_p++) {
+        if ((*parts_p)->start <= position && ((*parts_p)->start + (*parts_p)->size) > position) {
+            if ((*parts_p)->type == BD_PART_TYPE_EXTENDED) {
+                /* we don't want to return extended partition here -- there
+                   is either another logical or free space at this position */
+                bd_part_spec_free (*parts_p);
+                continue;
+            }
 
-    sector = (PedSector) (position / dev->sector_size);
-    ped_part = ped_disk_get_partition_by_sector (ped_disk, sector);
-    if (!ped_part) {
-        if (set_parted_error (error, BD_PART_ERROR_FAIL))
-            g_prefix_error (error, "Failed to get partition at position %"G_GUINT64_FORMAT" (device '%s')",
-                            position, disk);
-        else
-            /* no such partition, but no error */
-            g_clear_error (error);
-        ped_disk_destroy (ped_disk);
-        ped_device_destroy (dev);
-        return NULL;
+            ret = *parts_p;
```

Comment:
I think this should `break` or `!ret` should be added to the `for` loop condition above.

Reply:
Fixed.